server/src/code-runner/docker-runner.ts

```diff
@@ -48,40 +74,46 @@ export class DockerRunner implements CodeRunner {
 
     const mounts = flatMap(configuration.mountPoints, mp => ['-v', `${mp.source}:${mp.destination}:ro`]);
 
-    return this.config
-      .spawn(this.config.dockerExecutable, [
-        'create',
-        '--cap-drop=ALL',
-        `--kernel-memory=${this.config.maxKernelMemoryKb}k`,
-        '--security-opt=no-new-privileges',
-        '-t',
-        '--read-only',
-        '--tmpfs',
-        `/run:rw,size=${this.config.runPartitionSize},mode=${this.config.runPartitionMode}`,
-        '--tmpfs',
-        `/tmp:rw,size=${this.config.tmpPartitionSize},mode=${this.config.tmpPartitionMode}`,
-        ...mounts,
-        '-v',
-        `${tmpExecutionDir}:/lab:ro`,
-        `--name`,
-        invocation.id,
-        configuration.imageWithDigest,
-        '/bin/bash'
-      ])
+    return new UserResolver().resolve(invocation)
```

Comment:
Didn't find a better way (unless copying the exact same code as the UserResolver) to retrieve the user. Is this okay? Should we add a `UserModel` or something which can be used here and by the `UserResolver`?

Reply:
I think it would probably make sense to have a `UserModel` or `UserService` (similar to the one in the client) and then use it here and also inside `UserResolver`.

Unfortunately, I don't think it make a lot of sense to share the user service from the client, as this one always needs to authenticate. Although, we could implement an a DummyAuthService that always resolves. Then the service can be used on the server as well.

Reply:
What do you think would be the best approach here? Not all methods from the client `UserService` make sense on the server. So should we create a `SharedUserService` or something that can either be injected in the client (or maybe even extended from) and be used on the server.

Reply:
I think for what we need right now, we'd be better off just implementing a tiny UserService on the server. If it grows we can still move it into shared.

Reply:
Wow, thanks @cburgdorf. This is much easier and much cleaner :).